core/src/main/java/org/apache/accumulo/core/spi/scan/DefaultScanServerDispatcher.java

```diff
@@ -40,53 +42,241 @@
+ *         {"servers":"3", "busyTimeout":"30s","salt","42"},
+ *         {"servers":"9", "busyTimeout":"60s","salt","84"}
+ *       ]
+ *     }
+ *    ]
+ * </pre>
  *
+ * <p>
+ * For the default profile in the example it will start off by choosing randomly from 3 scan servers
+ * based on a hash of the tablet with no salt. For the first attempt it will use a busy timeout of
+ * 33 milliseconds. If the first attempt returns with busy, then it will randomly choose from 100%
+ * or all servers for the second attempt and use a busy timeout of 100ms. For subsequent attempts it
+ * will keep choosing from all servers and start multiplying the busy timeout by 4 until the max
+ * busy timeout of 4 minutes is reached.
+ * </p>
  *
+ * <p>
+ * For the profile activated by {@code scan_type=slow} it start off by choosing randomly from 1 scan
+ * server based on a hash of the tablet with no salt and a busy timeout of 10s. The second attempt
+ * will choose from 3 scan servers based on a hash of the tablet plus the salt {@literal 42}.
+ * Without the salt, the single scan servers from the first attempt would always be included in the
+ * set of 3. With the salt the single scan server from the first attempt may not be included. The
+ * third attempt will choose a scan server from 9 using the salt {@literal 84} and a busy timeout of
+ * 60s. The different salt means the set of servers that attempts 2 and 3 choose from may be
+ * disjoint. Attempt 4 and greater will continue to choose from the same 9 servers as attempt 3 and
+ * will keep increasing the busy timeout by multiplying 8 until the maximum of 20 minutes is
+ * reached.
+ * </p>
  */
 public class DefaultScanServerDispatcher implements ScanServerDispatcher {
 
   private static final Logger LOG = LoggerFactory.getLogger(DefaultScanServerDispatcher.class);
 
   private static final SecureRandom RANDOM = new SecureRandom();
+  public static final String PROFILES_DEFAULT = "[{'isDefault':true,'maxBusyTimeout':'5m',"
+      + "'busyTimeoutMultiplier':8, " + "'scanTypeActivations':[], "
+      + "'attemptPlans':[{'servers':'3', 'busyTimeout':'33ms', 'salt':'one'},"
+      + "{'servers':'13', 'busyTimeout':'33ms', 'salt':'two'},"
+      + "{'servers':'100%', 'busyTimeout':'33ms'}]}]";
 
-  protected Duration initialBusyTimeout;
-  protected Duration maxBusyTimeout;
+  private Supplier<List<String>> orderedScanServersSupplier;
 
-  protected int initialServers;
-  protected int maxDepth;
+  private Map<String,Profile> profiles;
+  private Profile defaultProfile;
 
-  private Supplier<List<String>> orderedScanServersSupplier;
+  private static final Set<String> OPT_NAMES = Set.of("profiles");
+
+  private static class AttemptPlan {
+    String servers;
+    String busyTimeout;
+    String salt = "";
+
+    transient double serversRatio;
+    transient int parsedServers;
+    transient boolean isServersPercent;
+    transient boolean parsed = false;
+    transient long parsedBusyTimeout;
+
+    void parse() {
```

Comment:
Suggest looking at GSON's TypeAdapterFactory to perform post-processing on the object. See https://www.javadoc.io/doc/com.google.code.gson/gson/2.6.2/com/google/gson/TypeAdapterFactory.html